torch/csrc/distributed/c10d/ParamCommsUtils.hpp

```diff
@@ -96,6 +96,7 @@ class TORCH_API ParamCommsDebugInfo : public c10::DebugInfoBase {
 
 #define RECORD_PARAM_COMMS(                                                    \
     seq,                                                                       \
+    isP2P,                                                                     \
```

Comment:
How about make seq be a tuple of (seq_id, is_p2p), but not a stand alone integer? Then make `RECORD_PARAM_COMM` and `RECORD_PARAM_COMM_DATA` have the same trace fromat, exept the first tensor. Otherwise, replayer needs to branch positional parsing.
After that, we will use `(seq_id, is_p2p, pg_id)` as key to match comm op and wait.